### PR TITLE
feat: add leaseCache to change gc policy

### PR DESCRIPTION
**Pull request by Desiki-high**
## Change GC policy

This patch changes GC policy from LRU to LFU/LRU: cleans the blob by LFU first then LRU.

Change:
- add singleflight for gc.
- add ```leaseCache```: the cache of lease, help record the used count and time.
- implement the new policy of GC by ```leaseCache```.
- update lease in commit. To make leaseCache is synchronized with local storage before read each blob.

⚠️Notice:
The ```leaseCache``` is **not thread-safe**, we make sure the safe by the ```gcSingleflight``` and the thread-safe of ```content.db.Update```.

Ref: https://github.com/goharbor/acceleration-service/issues/141

Test result of leaseCache:
- TestLeaseCache            0.007s Add(5) Remove(3)
- TestLeaseCacheInit 0.016s Add(6) Remove(1) Init(1)  Add (5)

### CODE REVIEW

File: pkg/content/content.go [hunk 1]
```diff
@@ -130,17 +138,16 @@ func (content *Content) GC(ctx context.Context) error {
...
-		return ls[i].Labels[usedAtLabel] < ls[j].Labels[usedAtLabel]
-	})
-	for _, lease := range ls {
+	for size > 0 {
+		if content.lc.Len() == 0 {
+			return fmt.Errorf("cleanLeases: leaseCache is empty, error caches")
+		}
+		digest, err := content.lc.Remove()
+		if err != nil {
+			return nil
```
Comment (imeoer) [inline pkg/content/content.go:147]: `return err`

File: pkg/content/cache_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+package content
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLeaseCahe(t *testing.T) {
+	lc := newLeaseCache()
```
Comment (imeoer) [inline pkg/content/cache_test.go:24]: Can we also add a perf test for `leaseCache` from `lc.Init` to `Add/Remove` ?

Comment (Desiki-high) [reply pkg/content/cache_test.go:24]: We can use temp ```content store```, ```db``` and ```leases Manager``` to test this.

File: pkg/content/cache_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	require.Equal(t, "test_1", key)
+	key, err = lc.Remove()
+	require.NoError(t, err)
+	require.Equal(t, "test_3", key)
+	key, err = lc.Remove()
+	require.NoError(t, err)
+	require.Equal(t, "test_2", key)
+}
+
+func TestLeaseCahePerformance(t *testing.T) {
```
Comment (imeoer) [inline pkg/content/cache_test.go:44]: How do we test the performance in this test case?

Comment (Desiki-high) [reply pkg/content/cache_test.go]: Rename to ```TestLeaseCacheInit```.

**Review by imeoer — APPROVED** [review]

**Pull request merged**